android/src/main/java/com/marigold/rnsdk/RNEngageBySailthruModule.kt

```diff
@@ -0,0 +1,347 @@
+package com.marigold.rnsdk
+
+import androidx.annotation.VisibleForTesting
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.ReadableArray
+import com.facebook.react.bridge.ReadableMap
+import com.marigold.sdk.EngageBySailthru
+import com.marigold.sdk.Marigold
+import com.marigold.sdk.model.AttributeMap
+import com.marigold.sdk.model.Purchase
+import org.json.JSONException
+import org.json.JSONObject
+import java.lang.reflect.InvocationTargetException
+import java.net.URI
+import java.net.URISyntaxException
+import java.util.ArrayList
+import java.util.Date
+
+class RNEngageBySailthruModule (reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
+
+    @VisibleForTesting
+    var engage = EngageBySailthru()
```

Comment:
This will still always attempt to create an instance of the `EngageBySailthru` class, which for non Sailthru customers will result in errors being thrown. I recommend creating a helper method which attempts to create an instance of `EngageBySailthru` when any methods are called then surfaces any exceptions, but that doesn't throw at the point the class is initialised. You can then use this in place of all your current calls to `engage`.